### ISSUE TITLE
semantic about softmax_cross_entropy_with_logits

### ISSUE THREAD

**Issue by playgithub** [tensorflow/tensorflow#8725]
In [https://www.tensorflow.org/get_started/mnist/beginners](url), it says:

> we call softmax_cross_entropy_with_logits on tf.matmul(x, W) + b)

In [https://www.tensorflow.org/api_docs/python/tf/nn/softmax_cross_entropy_with_logits](url), it says

> logits: Unscaled log probabilities.

Here is the question, take the following code for example:

```
z = tf.matmul(x, W) + b)
cost = tf.nn.softmax_cross_entropy_with_logits(logits=z, labels=the_lables)
```

**z** is not log probability, why pass it to the parameter **logits**?

**Comment by mckinziebrandon** [comment]
This belongs on StackOverflow (it has actually been asked many times on SO, see links below). __z__ is the log probability. That is a standard interpretation for the output of an affine transformation being fed to a softmax layer. For more, see:

* [SO: Difference between softmax and sofmtax with logits](http://stackoverflow.com/questions/34240703/difference-between-tensorflow-tf-nn-softmax-and-tf-nn-softmax-cross-entropy-with)
* [SO: tensorflow and unscaled log probabilities](http://stackoverflow.com/questions/40871797/tensorflow-softmax-cross-entropy-with-logits-asks-for-unscaled-log-probabilities)

**Comment by aselle** [comment]
As @mckinziebrandon  indicates, this question is better asked on  [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a  bug or feature request. There is also a larger community that reads questions there. Thanks!

**Comment by playgithub** [comment]
Thanks

**Comment by playgithub** [comment]
Because **softmax** can operate on any array, I think the concept **logits** is not necessary, and it'll be more easy.
